### ISSUE TITLE
Error with tf.losses.sparse_softmax_cross_entropy: weights?

### ISSUE THREAD

**Issue by Vonisoa** [tensorflow/tensorflow#12588]
I use the function tf.losses.sparse_softmax_cross_entropy for my unbalanced dataset but i get error when using it. My question is if my labels in one batch have shape (8,4096) , my prediction or logits have shape (8,4096,14) and I have 14 classes. What should the shape of my weights?

Another question: should i have one specific weight for each class or a specific weight for each sample?
Thanks

**Comment by ali01** [comment]
This question is better asked on  [StackOverflow](http://stackoverflow.com/questions/tagged/tensorflow) since it is not a  bug or feature request. There is also a larger community that reads questions there. Thanks!

**Comment by Vonisoa** [comment]
For those who will need it later, weight must have the same shape as the batch_size so (8,4096). Those weights represent weight for each sample.
